src/tools.rs

```diff
@@ -670,6 +670,16 @@ pub(crate) fn parse_receive_headers(headers: &Headers) -> String {
         .join("\n")
 }
 
+pub(crate) fn single_value<T>(collection: impl IntoIterator<Item = T>) -> Option<T> {
```

Comment:
For a function that is only used once that's a bit overkill isn't it?  Though if you do want to add this add a a full doc comment describing what it does - the naming is not as self-describing as you might think.